File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDODisplayModeParamModel.kt
//
//  IDODisplayModeParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Display mode event number

open class IDODisplayModeParamModel(mode: Int) : IDOBaseModel {

    /// Mode 
    /// 0: Default 
    /// 1: Landscape 
    /// 2: Portrait 
    /// 3: Flipped (180 degrees) 
    @SerializedName("mode")
    var mode: Int = mode


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDODisplayModeParamModel.swift
//
//  IDODisplayModeParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Display mode event number
@objcMembers
public class IDODisplayModeParamModel: NSObject, IDOBaseModel {
    /// Mode 
    /// 0: Default 
    /// 1: Landscape 
    /// 2: Portrait 
    /// 3: Flipped (180 degrees) 
    public var mode: Int
    
    enum CodingKeys: String, CodingKey {
        case mode
    }
    
    public init(mode: Int) {
        self.mode = mode
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

